server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4399,6 +4399,8 @@ private void persistVMDeployAsIsProperties(UserVmVO vm, Map<String, String> user
                             value = DBEncryptionUtil.encrypt(value);
                         }
                     }
+                } else if (value == null) {
+                    value = "";
```

Comment:
@nvazquez does this require special handling in the vmware resource/- guru? I mean to make sure vmware get the right type of value.

Reply:
how about not adding the key/value to map `userVmOVFPropertiesMap` when the value is null or empty?

Reply:
This could cause the deployments to fail as Vmware will expect some keys even with empty values

Reply:
@nvazquez Ok, In case VMware expects keys with empty values. Have you tested without the keys?

Reply:
> @nvazquez Ok, In case VMware expects keys with empty values. Have you tested without the keys?

@nvazquez Have you tested without the keys (instead of keys with empty values) ?

Reply:
Sorry @sureshanaparti I was on leave, no. Have not tested without the keys, will do